### PR TITLE
Redirect back to content from paywall in new window

### PR DESCRIPTION
**Pull request by cellog**
# Description

This is probably the last step on the path to #1287, pending testing in coinbase wallet

This PR adds the missing redirect from the paywall in a new window back to the main window, both when the user dismisses the modal, and when returning to the paywall after a key has been purchased. In the latter case, it only redirects if it finds a non-expired key.

*brief description of the screencast:*
In this screencast, you will observe me changing the default demo URL to instead go directly to the paywall. Then, using a handy-dandy url encoding website, I encoded the demo URL that the paywall should return to, and popped it on the end of the url. This is how the return-to URL is passed to the paywall.  Then, I purchase a key, and click "go to content" and it redirects. Afterwards, I return to the paywall URL with the url-encoded redirect pasted on the end, and you can see that it returns to the content as well. In addition, you can see that it successfully appended the user account to the content url as a hash, which will trigger passing it to the paywall in the iframe, and then trigger retrieval of keys in wallet browsers that do not have the wallet available in the iframe.

out

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [X] If my code involves visual changes, I am adding applicable screenshots to this thread

PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

File: unlock-app/src/middlewares/interWindowCommunicationMiddleware.js [hunk 1]
```diff
@@ -1,67 +1,105 @@
...
+        }
+      } else {
+        if (action.type === HIDE_MODAL) {
+          // if the user clicks the button to go to content,
+          // we redirect back to the content, appending as a hash
+          // the user account. This is not sent to the server.
+          // then, the paywall.min.js script detects the hash
+          // and forwards it to the paywall in the iframe.
+          // the code at the top of this file handles that case
+          redirectToContentFromPaywall(window, getState)
```
Comment (benwerd) [inline unlock-app/src/middlewares/interWindowCommunicationMiddleware.js]: Can this be repurposed for other Unlock applications later? It seems like it doesn't _need_ to be from the paywall only, right? Perhaps `redirectToContent` or `redirectToContentFromInterstitial`?

Comment (cellog) [reply unlock-app/src/middlewares/interWindowCommunicationMiddleware.js]: sure, I don't see why not. Did you want me to rename it? For now, the name is pretty accurate :)

**Comment by cellog** [comment]
Screenshots from Coinbase Wallet:

screenshot_20190208-151855
screenshot_20190208-160852
screenshot_20190208-161425
screenshot_20190208-161434
screenshot_20190208-161445
screenshot_20190208-161519
screenshot_20190208-161827

**Comment by cellog** [comment]
don't merge before #1488 and #1487

**Comment by cellog** [comment]
this is ready for final review, rebased and hungry

**Pull request merged**